var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -85,8 +92,12 @@ class Petsc(Package):
     # conflict in headers see
     # https://bitbucket.org/petsc/petsc/src/90564b43f6b05485163c147b464b5d6d28cde3ef/config/BuildSystem/config/packages/hypre.py
     depends_on('hypre~internal-superlu', when='+hypre+mpi~complex')
-    depends_on('superlu-dist@:4.3', when='@:3.6.4+superlu-dist+mpi')
+    depends_on('superlu-dist@:4.3', when='@3.6.4+superlu-dist+mpi')
     depends_on('superlu-dist@5.0.0:', when='@3.7:+superlu-dist+mpi')
+    # could spack be smart enough to know that master is greater than or equal to the 
+    #    to any release and thus not need the line below ? Or is there a better way to handle this?
+    #    if I don't add the line below and build with petsc master, it uses superlu-dist@4.3 which fails
```

Comment:
There is a [huge discussion](https://github.com/LLNL/spack/issues/1997) on this if you want to weigh in.  FWIW, we currently do this for `develop`.

My issue w/this is that `develop` (or `master`) is newer *now* but if we write that into the provenance and compare against it in the future, it's probably wrong (i.e., the installed `develop` version will one day be older than the most recent version).  My idea was to be smart enough to figure out what commit/recently tagged version it is *actually* at and write that into the provenance, so we can compare later.